tools/torch-mlir-opt/CMakeLists.txt

```diff
@@ -16,3 +16,7 @@ target_link_libraries(torch-mlir-opt PRIVATE
   ${dialect_libs}
   ${conversion_libs}
 )
+
+if(TORCH_MLIR_ENABLE_STABLEHLO)
+  target_link_libraries(torch-mlir-opt PRIVATE MhloDialect AllMhloPasses)
```

Comment:
Let's keep the dependencies to a minimum instead of linking `AllMhloPasses`.  If I understand correctly, the new pass you added (`TestUnfuseBatchNormPass`) is defined in mhlo/transforms/unfuse_batch_norm)/unfuse_batch_norm_pass.cc, which should be included in the `MhloPasses` library, which we link against in [`lib/CMakeLists.txt`](https://github.com/llvm/torch-mlir/blob/main/lib/CMakeLists.txt#L25-L29).

Which symbols do we miss if we don't link `MhloDialect` and `AllMhloPasses`?

Reply:
`mhlo::registerAllMhloPasses()` has passes more than `MhloPasses` library. So if we register it in `torch-mlir-opt`, we must link `AllMhloPasses`. 
Another method is only register `TestUnfuseBatchNormPass` in `torch-mlir-opt`. BTW, we need register all mhlo passes registered in `InitAll.cpp`, including `registerSymbolicShapeOptimizationPass` and so on. Otherwise we couldn't reproduce the error from `torch-mlir-opt`.

Reply:
I choose the second way. Please review again, thanks! @ashay